actors/eam/src/lib.rs

```diff
@@ -197,13 +171,13 @@ impl EamActor {
         rt.validate_immediate_caller_type(std::iter::once(&Type::Init))
     }
 
-    pub fn create<BS, RT>(rt: &mut RT, params: CreateParams) -> Result<RawBytes, ActorError>
+    pub fn create<BS, RT>(rt: &mut RT, params: CreateParams) -> Result<CreateReturn, ActorError>
     where
         BS: Blockstore + Clone,
         RT: Runtime<BS>,
     {
         rt.validate_immediate_caller_type(iter::once(&Type::EVM))?;
-        assert_code_size(&params.initcode)?;
+        rt.validate_immediate_caller_namespace(vec![&EAM_ACTOR_ADDR])?;
```

Comment:
I don't think we need this.

1. We end up checking below in `get_caller_address` anyways. We can (and probably should) just return forbidden there).
2. The EVM actor asserts this on construction.

Also, generally we're not allowed to "validate" twice (mostly just convention, but it's something we tend to enforce).